build.sbt

```diff
@@ -89,155 +74,191 @@ lazy val wallet = (project in file("modules/wallet"))
       Libraries.catsEffect,
+      Libraries.chill,
+      Libraries.circeCore,
+      Libraries.circeGeneric,
+      Libraries.circeParser,
+      Libraries.circeRefined,
       Libraries.cirisCore,
+      Libraries.cirisEnum,
+      Libraries.cirisRefined,
+      Libraries.comcast,
       Libraries.declineCore,
       Libraries.declineEffect,
       Libraries.declineRefined,
-      Libraries.fs2IO,
-      Libraries.refinedCore,
-      Libraries.refinedCats,
+      Libraries.derevoCore,
+      Libraries.derevoCats,
+      Libraries.derevoCirce,
+      Libraries.derevoScalacheck,
+      Libraries.doobieQuill,
+      Libraries.enumeratumCore,
+      Libraries.enumeratumCirce,
+      Libraries.guava,
       Libraries.log4cats,
       Libraries.logback % Runtime,
-      Libraries.logstashLogbackEncoder % Runtime
+      Libraries.logstashLogbackEncoder % Runtime,
+      Libraries.mapref,
+      Libraries.monocleCore,
+      Libraries.monocleMacro,
+      Libraries.newtype,
+      Libraries.refinedCore,
+      Libraries.refinedCats,
+      Libraries.refinedScalacheck,
+      Libraries.http4sCore
     )
   )
 
-lazy val keytool = (project in file("modules/keytool"))
-  .enablePlugins(AshScriptPlugin)
-  .dependsOn(shared, testShared % Test)
+lazy val testShared = (project in file("modules/test-shared"))
+  .configs(IntegrationTest)
   .settings(
-    name := "tessellation-keytool",
+    name := "tessellation-test-shared",
     Defaults.itSettings,
     scalafixCommonSettings,
     commonSettings,
     commonTestSettings,
-    makeBatScripts := Seq(),
     libraryDependencies ++= Seq(
       CompilerPlugin.kindProjector,
       CompilerPlugin.betterMonadicFor,
       CompilerPlugin.semanticDB,
-      Libraries.bc,
-      Libraries.bcExtensions,
+      Libraries.catsLaws,
+      Libraries.log4catsNoOp,
+      Libraries.monocleLaw,
+      Libraries.refinedScalacheck,
       Libraries.cats,
       Libraries.catsEffect,
       Libraries.circeCore,
       Libraries.circeGeneric,
       Libraries.circeParser,
       Libraries.circeRefined,
-      Libraries.cirisCore,
-      Libraries.comcast,
-      Libraries.derevoCore,
-      Libraries.derevoCats,
-      Libraries.derevoCirce,
-      Libraries.fs2IO,
-      Libraries.log4cats,
-      Libraries.logback % Runtime,
-      Libraries.logstashLogbackEncoder % Runtime,
-      Libraries.monocleCore,
-      Libraries.monocleMacro,
-      Libraries.newtype,
-      Libraries.declineCore,
-      Libraries.declineEffect,
-      Libraries.declineRefined,
-      Libraries.refinedCore,
-      Libraries.refinedCats
+      Libraries.fs2Core,
+      Libraries.http4sDsl,
+      Libraries.http4sServer,
+      Libraries.http4sClient,
+      Libraries.http4sCirce,
+      Libraries.http4sJwtAuth,
+      Libraries.weaverCats,
+      Libraries.weaverDiscipline,
+      Libraries.weaverScalaCheck
     )
   )
 
-lazy val shared = (project in file("modules/shared"))
+lazy val schema = (project in file("modules/schema"))
```

Comment:
We should have an ADR describing what's the purpose of each module so that we don't mix them up when adding stuff (like we do currently).